src/core/namespace.c

```diff
@@ -665,40 +674,45 @@ static int make_read_only(BindMount *m, char **blacklist) {
 
 static int chase_all_symlinks(const char *root_directory, BindMount *m, unsigned *n) {
         BindMount *f, *t;
-        int r;
+        int r = 0;
 
         assert(m);
-        assert(n);
+
+        /* Nothing to do */
+        if (!*n)
+                return r;
 
         /* Since mount() will always follow symlinks and we need to take the different root directory into account we
          * chase the symlinks on our own first. This call wil do so for all entries and remove all entries where we
          * can't resolve the path, and which have been marked for such removal. */
 
         for (f = m, t = m; f < m + *n; f++) {
                 _cleanup_free_ char *chased = NULL;
-
                 r = chase_symlinks(f->path, root_directory, &chased);
                 if (r == -ENOENT && f->ignore) {
                         /* Doesn't exist? Then remove it! */
                         f->path = mfree(f->path);
+                        r = 0;
                         continue;
                 }
-                if (r < 0)
-                        return log_debug_errno(r, "Failed to chase symlinks for %s: %m", f->path);
+                if (r < 0) {
+                        log_debug_errno(r, "Failed to chase symlinks for %s: %m", f->path);
+                        break;
+                }
```

Comment:
Correct, but it doing things this way leaks memory when the loop iteration is cut short...